geometry/create-and-edit-geometries/src/main/java/com/esri/samples/create_and_edit_geometries/CreateAndEditGeometriesController.java

```diff
@@ -0,0 +1,591 @@
+    // add new graphic to the graphics overlay
+    graphicsOverlay.getGraphics().add(graphic);
+  }
+
+  /**
+   * Updates a graphic already in the graphics overlay.
+   */
+  private void updateSelectedGraphic() {
+    // update the geometry of the graphic being edited and make it visible again
+    selectedGraphic.setGeometry(geometryEditor.stop());
+    selectedGraphic.setVisible(true);
+    selectedGraphic = null;
+  }
+
+  /**
+   * Stops the geometry editor without saving the geometry stored within.
+   */
+  @FXML
+  private void handleStopAndDiscardButtonClicked() {
+    // if editing a pre-existing graphic, make it visible again
+    if (selectedGraphic != null) {
+      selectedGraphic.setVisible(true);
+      selectedGraphic = null;
+      toolComboBox.setDisable(false);
+    }
+    geometryEditor.stop();
+  }
+
+  /**
+   * Removes all geometries from the graphics overlay.
+   */
+  @FXML
+  private void handleDeleteAllGeometriesButtonClicked() {
+    graphicsOverlay.getGraphics().clear();
+  }
+
+  /**
+   * Allows the user to select a graphic from the graphics overlay and edit its geometry.
+   */
+  private void selectGraphic() {
+
+    mapView.setOnMouseClicked(e -> {
+      graphicsOverlay.clearSelection();
+      Point2D mapViewPoint = new Point2D(e.getX(), e.getY());
+
+      // get graphics near the clicked location
+      ListenableFuture<IdentifyGraphicsOverlayResult> identifyGraphics;
+      identifyGraphics = mapView.identifyGraphicsOverlayAsync(graphicsOverlay, mapViewPoint, 10, false);
+
+      identifyGraphics.addDoneListener(() -> {
+        try {
+          // only select graphic if user not already creating/editing a geometry
+          if (!geometryEditor.isStarted()) {
+            if (!identifyGraphics.get().getGraphics().isEmpty()) {
+
+              // store the selected graphic
+              selectedGraphic = identifyGraphics.get().getGraphics().get(0);
+              selectedGraphic.setSelected(true);
+
+              // Use vertex tool if selected geometry is a point or multipoint
+              if (selectedGraphic.getGeometry().getGeometryType().equals(GeometryType.POINT) ||
+                selectedGraphic.getGeometry().getGeometryType().equals(GeometryType.MULTIPOINT)) {
+                geometryEditor.setTool(vertexTool);
+                toolComboBox.getSelectionModel().select(0);
+                toolComboBox.setDisable(true);
+              }
+
+              // hide the selected graphic & start an editing session with a copy of it
+              geometryEditor.start(selectedGraphic.getGeometry());
+              selectedGraphic.setVisible(false);
+            } else {
+              selectedGraphic = null;
+            }
+          }
+        } catch (Exception x) {
+          // on any error, display the stack trace
+          x.printStackTrace();
+        }
+      });
+    });
+  }
+
+  /**
+   * Defines graphics to be loaded into the sample upon startup.
+   */
+  void createInitialGraphics() {
```

Comment:
> Definition and loading of the example geometries is less than concise. I'm not particularly enthused about declaring them in the code like this, or converting them into List<double[]> to make their conversion into Points easier. Would it be better to move these into a JSON file and (after registering a type adapter) load them into a List<List<Double>>? I'm not sure there's any good option here.

Yes go ahead. See Display Device Location with Autopan Modes for an example of us following this approach in another sample. The key here is that the purpose of this sample is to create and edit geometries using the geometry editor, so obfuscating the creation of these initial graphics is ok.